codemods/buttons.js

```diff
@@ -0,0 +1,63 @@
+const visit = require('unist-util-visit');
+const {
+  removeAttribute,
+  addAttribute,
+  hasClassName,
+  findAttribute,
+} = require('./utils/mdxast');
+
+const VARIANTS = ['btn-primary'];
+
+const buttons = () => (tree, file) => {
+  if (file.path === './src/content/docs/agents/go-agent/index.mdx') {
+    console.dir(tree, { depth: 4 });
+  }
+  let addImport = false;
+  visit(
+    tree,
+    (node) => {
+      return hasClassName('btn', node);
+    },
+    (a) => {
+      addImport = true;
+      const className = findAttribute('className', a);
+      const href = findAttribute('href', a);
+
+      const variant = className
+        .split(/\s+/)
+        .find(
+          (className) =>
+            className.startsWith('btn-') && VARIANTS.includes(className)
+        )
+        .replace('btn-', '');
+
+      if (href) {
+        a.attributes.push({
+          name: 'as',
+          value: {
+            type: 'mdxValueExpression',
+            value: 'Link',
+          },
+          type: 'mdxAttribute',
+        });
+        addAttribute('to', href, a);
+      }
+
+      a.name = 'Button';
+      addAttribute('variant', variant, a);
+      removeAttribute('className', a);
+      removeAttribute('href', a);
+    }
+  );
+
+  // We are assuming all pages have frontmatter
```

Comment:
```suggestion
  // We are assuming all pages have frontmatter and therefore not handling
  // pages where frontmatter is missing. If the need arises, we'll have to insert
  // the import statement as the first child instead of the second child.
```

Added a little more to this comment to explain what needs to happen in case we need to handle cases with pages without frontmatter.
Adde